### PR TITLE
Template Animation: Added spin animations

### LINKED ISSUE #618: Basic Template Animations
## Feature Description

As an author I want to be able to use animations from a template

**[UX](https://www.figma.com/file/RTzSw47oi6pEZXEfP2JoBo/Google-Stories---Templates-(Copy)?node-id=1%3A5415)**

**Awaiting on feature brief for Animations to be written**

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria

1. As an author I am able to save a Bounce animation as a template animation
2. As an author I am able use a previously saved Bounce template animation
1. As an author I am able to save a Text Border animation as a template animation
2. As an author I am able use a previously saved Text Border template animation
1. As an author I am able to save a Text March animation as a template animation
2. As an author I am able use a previously saved Text March template animation
1. As an author I am able to save a Text Border March animation as a template animation
2. As an author I am able use a previously saved Text Border March template animation
1. As an author I am able to save a Text Reveal animation as a template animation
2. As an author I am able use a previously saved Text Reveal template animation
1. As an author I am able to save a Spin animation as a template animation
2. As an author I am able use a previously saved Spin template animation

### PR DESCRIPTION
**Pull request by mariano-formidable**
This PR adds `spin` animations to our repertoire.  

I also refactored how our animation configs work.  Instead of returning objects, I'm changing the defaults to return methods (which then return objects).   This way we can support dynamically generated keyframes (if needed) instead of only static ones.

The below animations are in our storybook and can be found at: 
`Dashboard -> Animations -> Spin`

Added animations:
Rotate:
default

Rotate counter-clockwise:
default-counter

Fast to slow:
fast-to-slow

Fast to slow counter-clockwise:
fast-to-slow-counter

Ping-pong:
ping-pong

### CODE REVIEW

File: assets/src/dashboard/animations/constants.js [hunk 1]
```diff
@@ -16,4 +16,11 @@
 
 export const ANIMATION_TYPE = {
   Bounce: 'bounce',
+  Spin: 'spin',
+};
+
+export const SPIN_TYPE = {
+  Clockwise: 'clockwise',
+  CounterClockwise: 'counterClockwise',
+  PingPong: 'pingPong',
```
Comment (BrittanyIRL) [inline assets/src/dashboard/animations/constants.js]: nit: consistent uppercase for all constants. In dashboard/constants we're doing all uppercase (SPIN_TYPE.CLOCKWISE) - we should pick one or the other and be consistent.

**Review by BrittanyIRL — APPROVED** [review]
These are so fun to play with! 

It would be great if you could add some knobs so that you can play with the settings in storybook.

File: assets/src/dashboard/animations/constants.js (hunk 1 above)
Comment (mariano-formidable) [reply assets/src/dashboard/animations/constants.js]: Good catch! I'll update the constants I'm using to animations to follow the all uppercase standard! 👍

**Comment by mariano-formidable** [comment]
> These are so fun to play with!
> 
> It would be great if you could add some knobs so that you can play with the settings in storybook.

Yeah, I want to add them in but unfortunately it looks like `amp-animation` (the scripts we're leveraging to animate all the things) caches the keyframe data on load.  So even if you change the data with knobs, the animation won't actually reload.  I tried making the clockwise and counter clockwise animations switchable by using a boolean knob but... it wouldn't reload the animations even as the keyframes changed :/ 

I am still new to `amp-animation`, so it's possible there _is_ a way to do this, and if/when I figure it out as I'm making more and more animations, I'll definitely reverse update and add in some knobs to make these more interactive :)

**Pull request merged**